tools/mason/MasonPublish.chpl

```diff
@@ -38,52 +38,28 @@ proc masonPublish(args: [] string) throws {
       masonPublishHelp();
       exit(0);
     }
-    if args.size == 4 {
-      masonPublishHelp();
-      exit(0);
-    }
     var dry = false;
+    var path = '';
     var username = getUsername();
-    if username == '' {
-      throw new owned MasonError('Must have remote origin set up in repository to publish');
+    var isLocal = false;
+
+    for arg in args[2..] {
+      if arg == '--dry-run' then dry = true;
+      else path = arg;
     }
-    else if args.size == 3 || args.size == 2 {
-      for arg in args[1..] {
-        if arg == '-h' || arg == '--help' {
-          masonPublishHelp();
-          exit(0);
-        }
-        else if arg == "--dry-run" {
-          dry = true;
-        }
-      }
-      if doesGitOriginExist() {
-        if dry && args.size == 3 {
-          dryRun(username);
-        }
-        else if !dry && args.size == 3 {
-          const badSyntaxMessage = ' does not meet the "mason publish [options]" syntax';
-          writeln('"' + args[0]+ ' '  + args[1] + ' ' + args[2] + badSyntaxMessage);
-          writeln('See "mason publish -h" for more details');
-          exit(0);
-        }
-        else if args.size == 2 {
-          var checkResult = usernameCheck(username);
-          if checkResult  == 0 {
-            publishPackage(username);
-          }
-          else {
-            const badUsernameError = 'mason-registry is not forked or ' + username + ' is not a valid username \n';
-            const badForkError = 'Make sure you are connected to the internet before publishing.';
-            throw new owned MasonError(badUsernameError + badForkError);
-          }
-        }
-        else if args.size == 4 && dry {
-          throw new owned MasonError('Must pass your username with --dry-run');
-        }
+    if path.isEmpty() {
+      path = MASON_HOME;
```

Comment:
Do we want to default to `MASON_HOME` or `MASON_REGISTRY`?

Reply:
I think `MASON_HOME` as we also have to access `/tmp` within `MASON_HOME` as well as clone the mason registry into .mason or wherever `MASON_HOME` points